server/src/domain/game.rs

```diff
@@ -4,10 +4,18 @@ use super::player::Player;
 // Types //
 ////////////////////////////////////////////////////////////////
 
-pub struct Game {
+pub enum Game {
+    Lobby(LobbyModel),
+}
+
+pub struct LobbyModel {
+    metadata: Metadata,
+    host: Player,
```

Comment:
What does it mean to be a host ? 

Can he kick other players out ? Does he get full control over the game state or what?  

I’m thinking a game shouldn’t have host since anyone can join it. Like when a game is created it automatically has a host id.

Do we want the functionality to kick players out or what ?

Reply:
I havent thought too much about all that. My motivation is that, the UX requires that _someone_ needs to start the game, and configure it (game name, game type, number of players, etc). So, one of the players needs to be differentiated as the one who can do those things. Furthermore _there must be exactly one_ player of this sort. If we just had `players: Vec<Player>` then we have to consider the case that the vector is empty, and that should be impossible. Also, if the player itself had some field like `is_host: bool`, then its possible in the data type that there would be two hosts- which also should be impossible. We want to "make impossible states impossible".